### DISCUSSION TITLE
Monochromatic icon variant

### DISCUSSION THREAD

**Discussion by sisp** [squidfunk/mkdocs-material#6304] [General]
I'd like to add Material for MkDocs' icon to [Simple Icons](https://simpleicons.org), which is a collection of monochromatic brand icons. Simple Icons, for example, provides the icons for shields.io badges and is also [available in MkDocs for Material](https://github.com/squidfunk/mkdocs-material/issues/4304#issuecomment-1239687743).

Adding Material for MkDocs' icon to Simple Icons could be useful for, e.g., advertising its use on GitHub/GitLab/... repos via a badge like

![Material for MkDocs](https://img.shields.io/badge/-Material%20for%20MkDocs-526cfe?style=for-the-badge&logo=readthedocs&logoColor=white)

but, of course, with the right icon (I chose Read the Docs as an example).

To my knowledge, Material for MkDocs has no official monochromatic icon, so I'd like to get some feedback which monochromatic variant you like &ndash; perhaps it could be added to this Git repo as well.

These are some ideas for monochromatic icons that I've come up with:

| Icon | Description |
| - | - |
| <img src="https://github.com/squidfunk/mkdocs-material/assets/2206639/b1b1d162-f8b3-4ec4-ba3d-88c9a115ecbd" width="128"/> | Shaded areas converted to strokes. |
| <img src="https://github.com/squidfunk/mkdocs-material/assets/2206639/577f41f0-f430-4c82-b2fd-ac721f7ab4dd" width="128"/> | Shaded areas converted to strokes but without strokes along the edges to the filled areas. IMO, looks a bit odd (strangely thin) at the bottom-most corner. |
| <img src="https://github.com/squidfunk/mkdocs-material/assets/2206639/c0544a2e-710c-452c-a2b7-1c908e7ab5f2" width="128"/> | Dark areas converted to strokes. |
| <img src="https://github.com/squidfunk/mkdocs-material/assets/2206639/ad6a44d6-6bc0-4de6-945c-85053f8f4263" width="128"/> | Dark areas converted to strokes but without strokes along the edges to the filled areas. |
| <img src="https://github.com/squidfunk/mkdocs-material/assets/2206639/91aabdc5-4d43-49d8-a3a2-a48437115f32" width="128"/> | All areas filled, gaps between them. |

Strokes and gaps are 0.5px thick on a 24x24px canvas. This is based on the recommendation in Simple Icons' [contribution guide](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#2-extract-the-icon-from-the-logo):

> Some icons can't be easily converted to a monochromatic version due to colour changes, shadows, or other effects. For such cases, the addition of gaps is the recommended approach, with a preferred width of 0.5px. In some situations, a different gap may be required, but that will be determined on a per-case basis.

I lean towards variant 1 and 3. WDYT, @squidfunk?

**Comment by squidfunk** [comment]
You are definitely one of our most awesome users ❤️ I love the idea of being part of Simple Icons, and since Material for MkDocs has become kind of a brand, I second the use cases for badges and stuff you mentioned.

Personally, I lean towards variant 4, as variant 3 and 4 most closely resemble our tinted version (white side on the front facing part of the shapes), and 3 looks a little odd because of the added strokes on the edges. Could I maybe ask you to make a variant with strokes only? I'd love to see it for comparison. I'd then try and put the logo in different contexts to decide ☺️

**Reply by sisp** [reply]
Thank you very much for the praise, I appreciate it! :pray::heart:

Here is a strokes-only variant:
<img src="https://github.com/squidfunk/mkdocs-material/assets/2206639/dbe4916d-1b05-44f2-8406-0ced8fcc1643" width="128" />

Variant 4 also looks good. :slightly_smiling_face: Let me know if you'd like to see more variants.

**Reply by alexvoss** [reply]
I imagine this would look good on a badge? How would be generate one? Here is [my attempt](https://badgen.net/#static/Material%20for%20MkDocs/9.4.8?icon=https://user-images.githubusercontent.com/2206639/281048280-dbe4916d-1b05-44f2-8406-0ced8fcc1643.svg) but I am not sure how to adapt the logo to be white on transparent? Is that something you would need to do in the image?

<img width="257" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/4134224/677a84eb-4aab-4276-bc25-f60b1a27aaae">

**Reply by sisp** [reply]
You could use Simple Icons' preview app: https://wasm.simpleicons.org/preview

The preview image looks like this for the strokes-only variant:
![materialformkdocs](https://github.com/squidfunk/mkdocs-material/assets/2206639/ed2790f3-95e1-46aa-a307-f411b0e00d43)

I added a white rectangle to the SVGs to make them visible also on GitHub in dark mode. When you upload them to the preview app, please remove the white rectangle first.

**Reply by squidfunk** [reply]
Sorry that it took me so long, I'm so busy with the search 😅 I played around with various combinations and feel that this version works best, since it has thicker outlines (so it works on small sizes):

``` svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M17.029 18.772l.777 1.166-5.417 2.709L0 16.451V4.063l5.417-2.709 5.298 7.948 7.867-5.24L24 1.354V16.84l-5.417 2.709-1.554-.777zm2.023-13.827v13.253l3.949-1.975V2.97l-3.949 1.975zM5.076 2.642L1.458 4.45 12.73 21.358l3.618-1.809L5.076 2.642z" />
</svg>
```

<img width="737" alt="Bildschirm­foto 2023-11-12 um 16 06 36" src="https://github.com/squidfunk/mkdocs-material/assets/932156/3f9a1460-542e-4e41-ac89-e8592c56a553">

I played around with moving the strokes (as you did), but it all looks off, at least to my gusto. Thus, I think just expanding the strokes and merging all shapes is the simplest and best working approach. What do you think? If you like it, would you like to create a PR over at SimpleIcons?

**Reply by sisp** [reply]
No worries! Sure, I'll submit the PR there tomorrow. Should we add it to this repo as well, so we have an official source URL?

**Reply by squidfunk** [reply]
Sure, here you go: 9ff75d118.

**Reply by sisp** [reply]
Perfect. One last question: What's your preferred brand color? `#526CFE` – the [text color from the animated color logo](https://github.com/squidfunk/mkdocs-material/blob/9ff75d11838cc01615697884c0eb9eb55f4652ad/.github/assets/logo.svg#L71)?

**Reply by squidfunk** [reply]
Yes, as indigo is our default color tone, I'd say we use `#526CFE`.

**Reply by sisp** [reply]
I've submitted a PR over at Simple Icons: https://github.com/simple-icons/simple-icons/pull/9851

**Reply by squidfunk** [reply]
Thanks for your work once again! I really love how this turned out – it's awesome that we now have a monochrome version ❤️

**Reply by sisp** [reply]
Sure thing! :slightly_smiling_face: I also like how it turned out. Wider strokes are definitely better for small icons like on shields.io badges. :ok_hand:
